File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {

        int start = 0;
        int end = nums.size() - 1;

        while(start < end) {
            int mid = start +(end - start)  / 2;
            if(nums[start] < nums[end]) return nums[start];
            if(nums[start] < nums[mid]) {
                start = mid;
            } else {
                end = mid;
            }
        }
        if(start == nums.size() - 1) return nums[start];
        return nums[start + 1];
      
    }
};

File: Data Structures & Algorithms/max-water-container/submission-7.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:


        lo = 0
        hi = len(nums) -1
        maxarea = 0

        while lo < hi:
            area = min(nums[lo], nums[hi]) * (hi - lo) # height times width
            maxarea = max(area, maxarea)
            if nums[lo] <= nums[hi]:
                lo += 1
            elif nums[lo] > nums[hi]:
                hi -= 1
        
        return maxarea




        